plugins/vbe/README.md

```diff
@@ -0,0 +1,326 @@
+  support device tree (e.g ACPI systems), this file is needed for VBE to work
+
+## Checking the state
+
+VBE stores its state in a file with the same name as the VBE method, with a
+`.dtb` extension. You can used the `fdtdump` utility to check the current state:
+
+```bash
+fdtdump /var/local/lib/fwupd/vbe/simple.dtb
+```
+
+## Available VBE methods
+
+At present only one method is available:
+
+* vbe,simple - writes a single copy of the firmware to media
+
+### vbe,simple
+
+With this, only a single copy of the firmware is available so if the write
+fails, the board may not boot. This is implemented in the vbe-simple.c file and
+has the device GUID ea1b96eb-a430-4033-8708-498b6d98178b within fwupd.
+
+Properties for this method are:
+
+* compatible - must be "fwupd,simple"
+* storage - device to store firmware in. Two options are supported: a full path
+   such as "/dev/mmcblk1" or a device number, like "mmc1". Note that only mmc
+   is currently supported.
+* area-start - start offset in bytes of the firmware area within the storage
+* area-size - size in bytes of the firmware area within the storage
+* skip-offset - offset to preserve at the start of the firmware area. This
+   means that the first part of the image is ignored, with just the latter part
+   being written. For example, if this is 0x200 then the first 512 bytes of the
+   image (which must be present in the image) are skipped and the bytes after
+   that are written to the store offset.
+* part-uuid - the UUID of the partition containing the fwupd state. This is not
+   used by fwupd at present but may allow the bootloader to check the fwupd
+   state on boot
+* part-id - the partition number containing the fwupd state. This is not
+   used by fwupd at present but may allow the bootloader to check the fwupd
+   state on boot
+
+State is typically stored in /var/local/lib/fwupd/vbe/simple.dtb in device tree
+format:
+
+```devicetree
+ / {
+     compatible = "vbe";
+     vbe-driver = "fwupd,simple";
+     last-update {
+         finish-time = <0x00000000 0x62a7429e>;
+         cur-version = "1.2.4";
+         status = "completed";
+     };
+```
+
+The fields are as follows:
+
+* compatible - the compatible string for VBE
+* vbe-driver - the actual VBE method being used
+* last-update - information about the last update
+* finish-time - the unix time of the last update, i.e. seconds since 1970, as a
+  64-bit value (0 if no previous update)
+* cur-version - the current installed version, missing if there is none or it is
+  not known
+* status - status of the update. The only supported value is 'completed'
+
+## Adding new VBE methods
+
+New VBE methods can be added in the plugin file `fu-plugin-vbe.c` in the
+`driver_list` array. See the documentation there for how to do this.
+
+## Finding out more
+
+See [fwupd site](https://github.com/fwupd/fwupd) for information about how
+to discuss or chat about fwupd.
```

Comment:
not sure this one make sense.  Isn't that where they got this from? :)

Reply:
removed